app/grouper_syncer.pl

```diff
@@ -0,0 +1,327 @@
+		log_info("User " . $grouper_user->{'id'} . " already exists in OESS but not this workgroup... adding to workgroup " . $workgroup->{'workgroup_id'});
+		$oess->add_user_to_workgroup( user_id => $all_oess_users{$grouper_user->{'username'}}->{'user_id'},
+					      workgroup_id => $workgroup->{'workgroup_id'});
+	    }else{
+		log_info("Adding user " . $grouper_user->{'username'} . " to OESS and to workgroup: " . $workgroup->{'workgroup_id'});
+		my $res = $oess->add_user( email_address => $grouper_user->{'email'},
+					   status => 'active',
+					   type => 'normal',
+					   family_name => $grouper_user->{'last_name'},
+					   first_name => $grouper_user->{'first_name'},
+					   auth_name => $grouper_user->{'username'});
+		log_debug("Add user result: " . Dumper($res));
+		if(!defined($res) || !defined($res->{'results'}->[0]->{'user_id'})){
+		    log_error("Error adding user to OESS: " . Dumper($res));
+		}else{
+		    my $user_id = $res->{'results'}->[0]->{'user_id'};
+		    $oess->add_user_to_workgroup( user_id => $user_id,
+						  workgroup_id => $workgroup->{'workgroup_id'});
+		}
+	    }
+	}
+    }
+
+    log_debug("Extra users that need removing: " . Dumper(keys %oess_users));
+
+    foreach my $ou (keys %oess_users){
+	log_info("Removing User: " . $ou . " from workgroup " . $workgroup->{'workgroup_id'});
+	$oess->remove_user_from_workgroup( workgroup_id => $workgroup->{'workgroup_id'},
+					   user_id => $oess_users{$ou}->{'user_id'} );
+	
+
+    }
+    
+
+}
+    
+sub get_grouper_users{
+    my %params = @_;
+    my $group = $params{'group'};
+    my $config = $params{'config'};
+    
+    my $url = $config->{'grouper'}->{'url'} . "/groups/" . $group->{'name'} . "/members";
+    
+    my $h = HTTP::Headers->new(
+        Content_Type => 'application/json' );
+    
+    my $r = HTTP::Request->new('GET', $url, $h);
+    $r->authorization_basic($config->{'grouper'}->{'user'}, $config->{'grouper'}->{'password'});
+    my $ua = LWP::UserAgent->new;
+    my $res = $ua->request($r);
+    
+    #log_debug("Results " . Dumper($res));
+    
+    my $obj = decode_json($res->content);
+    my @grouper_users;
+    
+
+    foreach my $gu (@{$obj->{'WsGetMembersLiteResult'}->{'wsSubjects'}}){
+	my $o = {'WsRestGetSubjectsRequest' => {'actAsSubjectLookup' => {'subjectId' => 'oess_provision'},
+						'wsSubjectLookups' => [{'subjectId' => $gu->{'id'}}],
+						'subjectAttributeNames' => ['givenname','sn','edupersonprincipalname','uid','employeenumber','cn','mail']}};
+	my $str = encode_json($o);
+	#log_debug("JSON STR: " . $str);
+#	my $str = '{
+#  "WsRestGetSubjectsRequest":{
+#    "actAsSubjectLookup":{
+#      "subjectId":"oess_provision"
+#    },
+#    "wsSubjectLookups":[
+#      {
+#        "subjectId":"' . $gu->{'id'} . '"
+#      }
+#    ] ,
+#    "subjectAttributeNames":[
+#      "givenname",
+#      "sn",
+#      "edupersonprincipalname",
+#      "uid",
+#      "employeenumber",
+#      "cn",
+#      "mail"
+#    ]
+#  }
+#}';
```

Comment:
Would probably make more sense to format the actual request and remove this comment?

Reply:
ya removed